### ISSUE TITLE
Goes for food inside a virus

### ISSUE THREAD

**Issue by kakol20** [Apostolique/Agar.io-bot#580]
When the bot is big enough, it sometimes goes for food inside a virus I've recently had an issue while using the bot, where it was trying to get a food in the centre of a virus and it split.

Maybe make a way for it to ignore food inside viruses.

**Comment by muratlahur** [comment]
This is a one-line code and I have fixed this about 6 months ago ... 

Only if the people in this project had some wits . . .

**Comment by muratlahur** [comment]
Hey, I've fixed this !!! it's a ONE-LINE code...

But the coders of this project can't get to that line..

instead they are busy deleting my posts here...

That's why your bot will keep eating food inside the viruses..

Sad, but true !

**Comment by Drflash55** [comment]
Yeah, that's something I suggested a while ago to be fixed. There should be a function that tells the bot that if it detects food or a cell inside of a virus, and it's not small enough to eat it without popping, it should ignore it entirely.

**Comment by ErmiyaEskandary** [comment]
Being worked on.

**Comment by ErmiyaEskandary** [comment]
This function checks if it is food or not - 
```
this.isFood = function(blob, cell) {
        if (!cell.isVirus() && this.compareSize(cell, blob, 1.33) || (cell.size <= 13)) {
            return true;
        }
        return false;
    };
```

**Comment by ErmiyaEskandary** [comment]
But it has no way of understanding if the food is **in** the virus - it only detects if it **isn't** a virus.

**Comment by muratlahur** [comment]
now that my acc is back online I guess you can safely jump to the fix ;)

**Comment by ErmiyaEskandary** [comment]
> brilliant geniuses who take care of this project will simply copy/paste my code in the bot like they always do

You are requesting for us to implement it - nobody asked for your help.

**Comment by ErmiyaEskandary** [comment]
@muratlahur if you stop acting like above, I will implement your code - I have no issue with you.

**Comment by muratlahur** [comment]
Hey man , truth .. Only the truth .. deal with it ..No seriously I mean absolutely no offense, it is what it is .. You'll copy  and paste that and it will work brilliantly.. 

I'm not helping because anyone asked, not doing for fame either.. I want people to have the best AI experience ..I have been developing bot since my botnet year back in the 90's ..

I love AI . ok .. And yet , very often I don't like humans..

so like I've said before.. please forget that I am a human , treat me as a bot..we'll go along just fine like that ..

**Comment by ErmiyaEskandary** [comment]
@Drflash55 @FliiFe Can you test ? I can't test due to where I'm at.

**Comment by Drflash55** [comment]
@ErmiyaEskandary Would love to, but my school firewall doesn't allow me to play Agar.io

Have to do it at home.

**Comment by muratlahur** [comment]
lol there is nothing to test . I am playing with it right now .. 

copy / paste at the lines I've indicated and you are good to go :)

**Comment by ErmiyaEskandary** [comment]
@muratlahur I can't edit the bot just because you say it works. I need someone else to test it as well, just to make sure. @FliiFe Can you ?

**Comment by ErmiyaEskandary** [comment]
@Drflash55 @Drflash55 @muratlahur I can test all of these in 40 minutes.

**Comment by muratlahur** [comment]
N-Joy :)

**Comment by ErmiyaEskandary** [comment]
@FliiFe Can you test this code?

**Comment by FliiFe** [comment]
@ErmiyaEskandary Yes, I will

**Comment by ErmiyaEskandary** [comment]
@FliiFe please post results here after testing.

**Comment by muratlahur** [comment]
KNOWN BUG... As this effectively deletes the food around viruses , if there are many viruses around and you are right in the middle of them , the bot kinda drifts as it has no food to eat until it reaches some food..

**Comment by muratlahur** [comment]
to avoid that ( and to get only the food under the virus deleted ) you should remove the * 5 ( this removes virus in a 5 x virus size diameter, if you delete that it will be just the virus size )

**Comment by FliiFe** [comment]
@ErmiyaEskandary I'm sorry to tell you there is a problem somewhere. The bot is definitely trying to work but can't manage to.

**Comment by FliiFe** [comment]
@ErmiyaEskandary Waiting for the bot to grow

**Comment by ErmiyaEskandary** [comment]
@FliiFe can you test now ?

**Comment by muratlahur** [comment]
ermiya , FliiFe has the entire source now .. I suggest you get it as well and just test and play with that as everything is where it should be

**Comment by muratlahur** [comment]
ESPECIALLY THE MULTICELL CONTROL JUST LOOK AT THE BRILLIANCE :D :D :D

**Comment by ErmiyaEskandary** [comment]
We don't want the whole source - we want what's new. Let's go step by step. FliiFe did the virus code work ? Let's start with this...

**Comment by ErmiyaEskandary** [comment]
Otherwise that's just copying your code.

**Comment by ErmiyaEskandary** [comment]
@FliiFe we think the same lol

**Comment by muratlahur** [comment]
ok , but my code  is apos bot + what's new .. so it's the same thing at the end lol .. 

if you merge all the new things , you'll end up with my code anyway hehe

**Comment by muratlahur** [comment]
anyhow..Look , I've given it all to you guys ....and I'm delighted that we understood each other on the ads and analytics..

so my job here is done really . . 

I'll be @ my mail when you need me .. 

Peeez out. . .

**Comment by muratlahur** [comment]
by the way , do study my code . there are many many many changes that I can't list..

From which food to chase , to how to handle threats to , re-unite and re-merge system to alone in the dark etc etc etc..

it will take you a while to completely understand what I did ..
